src/main/kotlin/com/amazon/opendistroforelasticsearch/indexstatemanagement/step/forcemerge/WaitForForceMergeStep.kt

```diff
@@ -112,7 +112,7 @@ class WaitForForceMergeStep(
             val statsResponse: IndicesStatsResponse = client.admin().indices().suspendUntil { stats(statsRequest, it) }
 
             if (statsResponse.status == RestStatus.OK) {
-                return statsResponse.shards.count { it.stats.segments.count > maxNumSegments }
+                return statsResponse.shards.count { it.stats.segments!!.count > maxNumSegments }
```

Comment:
Don't use !! in production code, using it in our tests is fine, this opens up possibility for NPEs

Reply:
Multiple places below too